aws-android-sdk-appsync/src/main/java/com/amazonaws/mobileconnectors/appsync/SubscriptionAuthorizer.java

```diff
@@ -38,17 +39,22 @@
  * Create one per {@link AWSAppSyncClient}.
  */
 class SubscriptionAuthorizer {
-    private final AWSConfiguration awsConfiguration;
-    private final Context applicationContext;
+    private final AWSConfiguration mAwsConfiguration;
+    private final Context mApplicationContext;
     private final OidcAuthProvider mOidcAuthProvider;
-
-    SubscriptionAuthorizer(
-            AWSConfiguration awsConfiguration,
-            OidcAuthProvider mOidcAuthProvider,
-            Context applicationContext) {
-        this.awsConfiguration = awsConfiguration;
-        this.applicationContext = applicationContext;
-        this.mOidcAuthProvider = mOidcAuthProvider;
+    private final AWSCredentialsProvider mCredentialsProvider;
+    private final String mRegionStr;
+    private final String mServerUrl;
+    private final APIKeyAuthProvider mApiKeyProvider;
+
+    SubscriptionAuthorizer(AWSAppSyncClient.Builder builder) {
+        this.mAwsConfiguration = builder.mAwsConfiguration;
+        this.mApplicationContext = builder.mContext;
+        this.mOidcAuthProvider = builder.mOidcAuthProvider;
+        this.mCredentialsProvider = builder.mCredentialsProvider;
+        this.mRegionStr = builder.mRegion.getName();
+        this.mServerUrl = builder.mServerUrl;
+        this.mApiKeyProvider = builder.mApiKey;
```

Comment:
Changing the constructor to just accept a builder, and pulling all of the necessary fields out of the builder to local members.